src/EnergyPlus/UnitarySystem.cc

```diff
@@ -2639,7 +2639,8 @@ namespace UnitarySystems {
 
             newCoil.size(state);
             if (MSHPIndex == -1) {
-                for (Iter = 1; Iter <= this->m_NumOfSpeedCooling; ++Iter) {
+                // Gotta loop backwards since we may try to access the last element when there are no fans
+                for (Iter = this->m_NumOfSpeedCooling; Iter >= 1; --Iter) {
```

Comment:
See here too

Reply:
Hmmm  @rraustad care to chime in here as well?

Reply:
Looks good to me. As the comment says, line 2650 will divide by the max speed air flow and it needs to be available. There are several loops that go in reverse order.